src/components/FloatingToolbar.tsx

```diff
@@ -82,35 +82,66 @@ function usePosition({ menuRef, isSelectingText, props }) {
     selectionBounds.right = selectionBounds.left = selectionBounds.left - 18;
   }
 
-  // calcluate the horizontal center of the selection
-  const halfSelection =
-    Math.abs(selectionBounds.right - selectionBounds.left) / 2;
-  const centerOfSelection = selectionBounds.left + halfSelection;
-
-  // position the menu so that it is centered over the selection except in
-  // the cases where it would extend off the edge of the screen. In these
-  // instances leave a margin
-  const margin = 12;
-  const left = Math.min(
-    window.innerWidth - menuWidth - margin,
-    Math.max(margin, centerOfSelection - menuWidth / 2)
-  );
-  const top = Math.min(
-    window.innerHeight - menuHeight - margin,
-    Math.max(margin, selectionBounds.top - menuHeight)
-  );
-
-  // if the menu has been offset to not extend offscreen then we should adjust
-  // the position of the triangle underneath to correctly point to the center
-  // of the selection still
-  const offset = left - (centerOfSelection - menuWidth / 2);
+  const isImageSelection =
+    selection.node && selection.node.type.name === "image";
+  // Images need their own positioning to get the toolbar in the center
+  if (isImageSelection) {
+    const { node: element } = view.domAtPos(selection.ranges[0].$to.pos);
+    const { width } = element.getBoundingClientRect();
 
-  return {
-    left: Math.round(left + window.scrollX),
-    top: Math.round(top + window.scrollY),
-    offset: Math.round(offset),
-    visible: true,
-  };
+    const isLeftAligned = selection.node.attrs.layoutClass === "half-left";
+    const isRightAligned = selection.node.attrs.layoutClass === "half-right";
+    let left = Math.round(element.offsetLeft + element.offsetParent.offsetLeft);
+
+    if (isLeftAligned) {
```

Comment:
![image](https://user-images.githubusercontent.com/380914/103451220-4573c900-4c76-11eb-967f-d40fd5a56f11.png)

This isn't quite centred right now

Reply:
Thanks, I finally figured it out, I needed to traverse into the element to the actual `div.image` and now the positioning works fine and is much simpler.